lite/examples/image_classification/android/lib_support/src/main/java/org/tensorflow/lite/examples/classification/tflite/Classifier.java

```diff
@@ -204,11 +206,19 @@ protected Classifier(Activity activity, Device device, int numThreads) throws IO
         tfliteOptions.addDelegate(nnApiDelegate);
         break;
       case GPU:
-        gpuDelegate = new GpuDelegate();
-        tfliteOptions.addDelegate(gpuDelegate);
-        break;
+        CompatibilityList compatList = new CompatibilityList();
+        if(compatList.isDelegateSupportedOnThisDevice()){
+          // if the device has a supported GPU, add the GPU delegate
+          GpuDelegate.Options delegateOptions = compatList.getBestOptionsForThisDevice();
+          GpuDelegate gpuDelegate = new GpuDelegate(delegateOptions);
+          tfliteOptions.addDelegate(gpuDelegate);
+          Log.d(TAG, "GPU supported. GPU delegate created and added to options");
+          break;
+        }
+        Log.d(TAG, "GPU not supported. Default to CPU.");
```

Comment:
Need a "break" in the end.

Reply:
This is a deliberate choice.

I was thinking by not having a break, it will flow to the CPU part without duplicating the code `tfliteOptions.setUseXNNPACK(true);` in an else branch. I tested this using the emulator to make sure if GPU compatibility is false, it will go through the CPU branch.

If you don't think the code is clear, I can put in a else branch and duplicate `tfliteOptions.setUseXNNPACK(true);`.

Thoughts?

Reply:
Aha, please put `tfliteOptions.setUseXNNPACK(true);` in the else branch. It makes a better readability. :-)

Reply:
Updated per suggestion in the latest commit.